### ISSUE TITLE
dotParse as a directive  part III 

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#11362]
In the quest for minimizing the memory usage of velocity, we need to migrate away from macros and move to directives

**Comment by wezell** [comment]
Here is the difference in the thread stack - this is just measured from the start of the VelocityServlet.doLive() render to the pull personalized footer of the starters /about-us/index page.  


**New**
```
    at com.dotmarketing.viewtools.content.ContentTool.pullPersonalized(ContentTool.java:433)
        at sun.reflect.GeneratedMethodAccessor450.invoke()
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:372)
        at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:361)
        at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:178)
        at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:285)
        at org.apache.velocity.runtime.parser.node.ASTReference.value(ASTReference.java:572)
        at org.apache.velocity.runtime.directive.Foreach.render(Foreach.java:321)
        at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:211)
        at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:339)
        at com.dotmarketing.velocity.directive.DotDirective.renderTemplate(DotDirective.java:114)
        at com.dotmarketing.velocity.directive.DotParse.renderTemplate(DotParse.java:1)
        at com.dotmarketing.velocity.directive.DotDirective.render(DotDirective.java:62)
        at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:211)
        at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
        at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
        at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:339)
        at org.apache.velocity.runtime.directive.Parse.render(Parse.java:264)
        at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:211)
        at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
        at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
        at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:339)
        at org.apache.velocity.Template.merge(Template.java:362)
        at org.apache.velocity.Template.merge(Template.java:265)
        at com.dotmarketing.velocity.VelocityServlet.doLiveMode(VelocityServlet.java:583)
```


**Old**
```

	at com.dotmarketing.viewtools.content.ContentTool.pullPersonalized(ContentTool.java:430)
	at sun.reflect.GeneratedMethodAccessor1121.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:372)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:361)
	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:178)
	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:285)
	at org.apache.velocity.runtime.parser.node.ASTReference.value(ASTReference.java:572)
	at org.apache.velocity.runtime.directive.Foreach.render(Foreach.java:321)
	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:211)
	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:330)
	at org.apache.velocity.Template.merge(Template.java:362)
	at org.apache.velocity.Template.merge(Template.java:265)
	at com.dotmarketing.viewtools.VelocityWebUtil.mergeTemplate(VelocityWebUtil.java:90)
	at sun.reflect.GeneratedMethodAccessor1049.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:372)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:361)
	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:178)
	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:285)
	at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:374)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:218)
	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:308)
	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:226)
	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:211)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:330)
	at org.apache.velocity.Template.merge(Template.java:362)
	at org.apache.velocity.Template.merge(Template.java:265)
	at com.dotmarketing.viewtools.VelocityWebUtil.mergeTemplate(VelocityWebUtil.java:90)
	at sun.reflect.GeneratedMethodAccessor1049.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:372)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:361)
	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:178)
	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:285)
	at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:374)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:330)
	at org.apache.velocity.Template.merge(Template.java:362)
	at org.apache.velocity.Template.merge(Template.java:265)
	at com.dotmarketing.velocity.VelocityServlet.doLiveMode(VelocityServlet.java:608)
```

**Comment by jtesser** [comment]
well i love directives and hate macros so I agree 100%

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/11382

**Comment by jgambarios** [comment]
**Note to QA:** The tests related to this ticket are going to be made in https://github.com/dotCMS/core/issues/11437

**Comment by bryanboza** [comment]
Fixed, tested on release-4.1 // Postgres // FF
